controllers/database_reconciler_helpers.go

```diff
@@ -260,11 +269,15 @@ func (r *DatabaseReconciler) handleSync(ctx context.Context, database *ndbv1alph
 				return r.addFinalizer(ctx, req, common.FINALIZER_DATABASE_SERVER, database)
 			}
 		}
-		r.setupConnectivity(ctx, database, req)
+		if databaseStatus.IPAddress != "" {
+			r.setupConnectivity(ctx, database, req)
+		}
 	case common.DATABASE_CR_STATUS_DELETING:
 		return r.handleDelete(ctx, database, ndbClient)
 	case common.DATABASE_CR_STATUS_NOT_FOUND:
 		r.recorder.Eventf(database, "Warning", EVENT_EXTERNAL_DELETE, "Error: Resource not found on NDB")
+	case common.DATABASE_CR_STATUS_CREATION_ERROR:
```

Comment:
Interesting, previously when we tried provisioning and there was something wrong with the yaml or provisioning request I believe it kept requeing? Now you'lll have it stop, nice, tracking the operation id is powerful.

Reply:
Yes, we can still introduce re-queuing for certain cases. But some states can be classified as non-recoverable error states where re-queuing might not help. We can add this framework later on.